Remove non-standard `WebSocketStream` from WebSocket API (#43654)
mdn/content: files/en-us/web/api/websockets_api/index.md

title: WebSocket API (WebSockets)
slug: Web/API/WebSockets_API
page-type: web-api-overview
browser-compat: api.WebSocket
---

{{DefaultAPISidebar("WebSockets API")}}{{AvailableInWorkers}}

The **WebSocket API** makes it possible to open a two-way interactive communication session between the user's browser and a server. With this API, you can send messages to a server and receive responses without having to poll the server for a reply.

The `WebSocket` interface is stable and has good browser and server support. However it doesn't support [backpressure](/en-US/docs/Web/API/Streams_API/Concepts#backpressure). As a result, when messages arrive faster than the application can process them it will either fill up the device's memory by buffering those messages, become unresponsive due to 100% CPU usage, or both.

> The [WebTransport API](/en-US/docs/Web/API/WebTransport_API) is expected to replace the WebSocket API for many applications. WebTransport is a versatile, low-level API that provides backpressure and many other features not supported by `WebSocket`, such as unidirectional streams, out-of-order delivery, and unreliable data transmission via datagrams. WebTransport is more complex to use than WebSockets and its cross-browser support is not as wide, but it enables the implementation of sophisticated solutions. If standard WebSocket connections are a good fit for your use case and you need wide browser compatibility, you should employ the WebSockets API to get up and running quickly. However, if your application requires a non-standard custom solution, then you should use the WebTransport API.

Reviewer: Maybe something like:

```suggestion
The [WebTransport API](/en-US/docs/Web/API/WebTransport_API) provides a useful alternative to the WebSocket API for many applications. WebTransport is a versatile, low-level API that provides backpressure and many other features not supported by `WebSocket`. WebTransport is more complex to use than WebSockets, but it enables the implementation of sophisticated solutions. If standard WebSocket connections are a good fit for your use case and you need wide browser compatibility, you should employ the WebSockets API to get up and running quickly. However, if your application requires a more complex solution and backpressure is an issue, then you should use the WebTransport API.
```

Cut down on the verbiage a bit that is available on the WebTransport page, and also try to cut down on the text that will age badly as WebTransport support improves.
